core/trino-main/src/main/java/io/trino/server/QueryResource.java

```diff
@@ -98,6 +100,11 @@ public Response getQueryInfo(@PathParam("queryId") QueryId queryId, @Context Htt
         }
         try {
             checkCanViewQueryOwnedBy(sessionContextFactory.extractAuthorizedIdentity(servletRequest, httpHeaders), queryInfo.get().getSession().toIdentity(), accessControl);
+
+            if (pruned) {
```

Comment:
instead of reassigning the variable we can do just:

```
Optional<QueryInfo> queryInfo = dispatchManager.getFullQueryInfo(queryId)
                .map(info -> pruned ? pruneQueryInfo(info, info.getVersion()) : info);
```

above